remotespark/wrapperkernel/sparkkernelbase.py

```diff
@@ -224,4 +238,4 @@ def _repeat_fatal_error(self):
         error = conf.fatal_error_suggestion().format(self._fatal_error)
         self._logger.error(error)
         self._ipython_display.send_error(error)
-        raise ValueError(self._fatal_error)
+        return self._complete_cell()
```

Comment:
To my knowledge, there are no unit tests that check the return value of `do_execute` -- all of them just that calling the function has the right behavior.